solr/test-framework/src/java/org/apache/solr/cloud/MiniSolrCloudCluster.java

```diff
@@ -1174,16 +1379,33 @@ public MiniSolrCloudCluster build() throws Exception {
       }
 
       JettyConfig jettyConfig = jettyConfigBuilder.build();
-      MiniSolrCloudCluster cluster =
-          new MiniSolrCloudCluster(
-              nodeCount,
-              baseDir,
-              solrXml,
-              jettyConfig,
-              null,
-              securityJson,
-              trackJettyMetrics,
-              formatZkServer);
+      MiniSolrCloudCluster cluster;
+
+      if (useEmbeddedZkQuorum) {
```

Comment:
this big if statement seems challenging...   Could a builder pattern make it easier?   Thir are a lot of params and overlap with subtle differences ;-).

Reply:
Hmm, this code **is** inside the builder :) So the builder chose to instantiate the MSCC using one of two constructors.